src/odemis/driver/xt_client.py

```diff
@@ -1506,15 +1536,26 @@ def _acquire(self):
                 while True:
                     if self._acq_should_stop():
                         break
-                    self.parent.set_channel_state(self._channel_name, True)
-                    if self.parent._scanner.blanker.value is None:
-                        self.parent.unblank_beam()
+                    self.parent.set_channel_state(self._scanner.channel, True)
+                    if hasattr(self._scanner, "blanker"):
+                        if self._scanner.blanker.value is None:
+                            self.parent.unblank_beam()
```

Comment:
Note that if the _scanner has changed since last image, we should blank the *previous* scanner. 
So you should keep a reference to the scanner, so blank the beam if it is changed. 
That said, it seems nothing of this is currently handled. So maybe  you could just add a TODO, and let it be done when someone fixes the blanker handling.